fence/agents/scsi/fence_scsi.py

```diff
@@ -182,7 +182,14 @@ def get_cluster_id(options):
 	cmd = options["--corosync-cmap-path"] + " totem.cluster_name"
 
 	match = re.search(r"\(str\) = (\S+)\n", run_cmd(options, cmd)["out"])
-	return hashlib.md5(match.group(1)).hexdigest() if match else fail_usage("Failed: cannot get cluster name")
+
+	if not match:
+		fail_usage("Failed: cannot get cluster name")
+
+	try:
+		return hashlib.md5(match.group(1), usedforsecurity=False).hexdigest()
+	except TypeError:
+		return hashlib.md5(match.group(1)).hexdigest()
```

Comment:
If you claimed "RHEL/CentOS using FIPS is a huge majority", that
would be something to take into account.  I highly doubt that's the case.
Furthermore, you must have missed that recent Fedoras do not support
`usedforsecurity`:

```
$ python -c 'import hashlib; print(hashlib.md5("dfdsf", usedforsecurity=False))'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: openssl_md5() takes no keyword arguments

$ python3 -c 'import hashlib; print(hashlib.md5(bytes("dfdsf", "ascii"), usedforsecurity=False))'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: openssl_md5() takes no keyword arguments
```